### ISSUE TITLE
Lab 6: Compilation warning and unable to pass case 8

### ISSUE THREAD

**Issue by BlabbyDuck** [nus-cs2030/2223-s1#195]
# Summary
Im not sure as to why this warning pops out 
<img width="945" alt="Screenshot 2022-10-14 at 11 31 13 PM" src="https://user-images.githubusercontent.com/108108396/195884793-39394b52-faa5-4a6d-8f60-bfcc7a960f5f.png">

## Description:

Im also stuck on case 8. Anyone have any idea as to why this error occur? 
<img width="427" alt="Screenshot 2022-10-14 at 11 48 52 PM" src="https://user-images.githubusercontent.com/108108396/195888235-9c0a0d4d-e546-443b-a577-1958951037f8.png">

**Comment by MingFengC** [comment]
It seems that your Server free time was updated incorrectly after taking a break, since the expected output shows that Customer 3 is served by Server 1 without a preceding `Done` event

**Comment by wilsonwid** [comment]
Aside from that, it seems there is a timing issue with regards to how a `Customer` should be served (notice that even though Customer 3 arrives earlier than 5, 5 gets served before 3, meaning the ordering of events is incorrect in the PQ).

**Comment by keefetok** [comment]
there seems to be a timing issue with your wait and serve events, maybe you can check how you are passing the time in the constructor around and fix from there (make your own test case to debug)

**Comment by BlabbyDuck** [comment]
Solved it!

**Comment by GarlicButterSteak** [comment]
@BlabbyDuck Hello! May I ask how you solved the issue for the 8th test case in your code? I have the same strange thing that you had where Customer 5 gets served before Customer 3 even though 3 arrived earlier 😔 but I can't seem to figure out why

![Bug](https://user-images.githubusercontent.com/77563792/198851358-38a46faa-6013-4aca-af68-cab00d348d8f.JPG)

**Comment by BlabbyDuck** [comment]
Hello! If I remember correctly this issue occurs when the server is done serving the previous customer and rest. If a new customer(cust 3 in this case) comes in and queues behind the server while he is resting, since my serve event is called after a done event, cust 3 will not be served until another customer (cust 5) is called. Hence cust 3 will keep waiting meanwhile regardless of his priority. 

What I did was in the simulator class, I made sure if the event was a wait class, if the server is resting and does not have any queue/customer, I will create a server event in the simulator class. 

Feel free to ask me more qns to clarify your doubts!
